### PR TITLE
Add homepage, todo page with Google sign in auth and API endpoints for todo

### PR DESCRIPTION
**Pull request by PratibhaShrivastav**

### CODE REVIEW

File: src/main/java/com/google/sps/data/ResponseStatus.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.google.sps.data;
+import lombok.Builder;
+
+@Builder
+public final class ResponseStatus{
+    private final int status_code;
+    private final String status_message;
```
Comment (yashjaing) [inline src/main/java/com/google/sps/data/ResponseStatus.java:7]: camel case for java

File: src/main/java/com/google/sps/servlets/MarkTodoServlet.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+/* Servlet that handles adding items to todo list */
+@WebServlet("/mark_todo")
+public class MarkTodoServlet extends HttpServlet {
+
+    private static final String PROFILE_ID = "ProfileID";
+    private static final String TODO_ENTITY = "TodoEntity";
+    private static final String ENTITY_TYPE_PROPERTY = "EntityType";
+    private static final String ENTITY_ID_PROPERTY = "EntityID";
+    private static final String TIMESTAMP_PROPERTY = "timestamp";
```
Comment (yashjaing) [inline src/main/java/com/google/sps/servlets/MarkTodoServlet.java:29]: same

File: src/main/java/com/google/sps/servlets/MarkTodoServlet.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    private static final Gson GSON = new Gson();
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        HttpSession session = request.getSession();
+        String profileID = (String)session.getAttribute("profileID");
+        int entityType = Integer.parseInt(request.getParameter("EntityType"));
+        long entityID = Integer.parseInt(request.getParameter("EntityID"));
```
Comment (yashjaing) [inline src/main/java/com/google/sps/servlets/MarkTodoServlet.java]: you already have the keys as string constants. use them

File: src/main/java/com/google/sps/servlets/ReviewServlet.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+public class ReviewServlet extends HttpServlet {
+
+    private static final String PROFILE_ID = "ProfileID";
+    private static final String REVIEW_ENTITY = "ReviewEntity";
+    private static final String PROFILE_ID_PROPERTY = "ProfileID";
+    private static final String ENTITY_TYPE_PROPERTY = "EntityType";
+    private static final String ENTITY_ID_PROPERTY = "EntityID";
+    private static final String ENTITY_RATING_PROPERTY = "Rating";
+    private static final String ENTITY_COMMENT_PROPERTY = "Comment";
+    private static final String TIMESTAMP_PROPERTY = "timestamp";
```
Comment (yashjaing) [inline src/main/java/com/google/sps/servlets/ReviewServlet.java]: looks like the same constants everywhere...can you please create a common/Constants.java and use that everywhere?

File: src/main/java/com/google/sps/servlets/TodoListServlet.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        HttpSession session = request.getSession();
+        String profileID = (String)session.getAttribute("profileID");
+        Filter propertyFilter = new FilterPredicate(PROFILE_ID_PROPERTY, FilterOperator.EQUAL, profileID);
+        Query query = new Query(TODO_ENTITY).setFilter(propertyFilter);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        List<BaseEntity> todo_list = new ArrayList<>();
```
Comment (yashjaing) [inline src/main/java/com/google/sps/servlets/TodoListServlet.java:48]: case :(

File: src/main/java/com/google/sps/servlets/UnmarkTodoServlet.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            long ID = (long)entity.getProperty(ENTITY_ID_PROPERTY);
+            long type = (long)entity.getProperty(ENTITY_TYPE_PROPERTY);
+            if ((ID == entityID) && (type == entityType))
+            {
+                Key entityKey = entity.getKey();
+                datastore.delete(entityKey);
+                break;
+            }
+        }
+        ResponseStatus responseStatus = ResponseStatus.builder().status_code(HttpServletResponse.SC_OK).status_message("OK").build();
```
Comment (yashjaing) [inline src/main/java/com/google/sps/servlets/UnmarkTodoServlet.java:58]: message is not required

File: src/main/webapp/assets/javascript/app.js [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+        codeHTML += '<div class="col-sm-6 moviePosterInModal">';
+        // codeHTML += '<a href="' + youtubeLink + '"><img src="' + poster + '"></a>';
+        codeHTML += '<img src="' + poster + '">';
+        codeHTML += '</div><br>'; //close trailerLink
+        codeHTML += '<div class="col-sm-6 movieDetails">';
+        codeHTML += '<div class="movieName">' + title + '</div><br>';
+        // codeHTML += '<div class="linkToTrailer"><a href="' + youtubeLink + '"><span class="glyphicon glyphicon-play"></span>&nbspPlay trailer</a>' + '</div><br>';
+        codeHTML += '<div class="release">Release Date: ' + releaseDate + '</div><br>';
+        codeHTML += '<div class="overview">' + overview + '</div><br>'; // Put overview in a separate div to make it easier to style
+        codeHTML += '<div class="rating">Rating: ' + voteAverage + '/10</div><br>';
```
Comment (yashjaing) [inline src/main/webapp/assets/javascript/app.js]: can this be a single string.. I am afraid so many concats will bring in in-efficiency

**Review by yashjaing — CHANGES_REQUESTED** [review]

File: src/main/webapp/assets/javascript/app.js (hunk 7 above)
Comment (anishajain22) [reply src/main/webapp/assets/javascript/app.js]: Noted. Modified.

**Pull request merged**